libs/providers/vmware.py

```
@@ -863,36 +862,36 @@ def _get_add_disk_device_specs(
             # Determine which datastore to use for this disk
             disk_datastore_id = disk.get("datastore_id")
-            LOGGER.info(f"Processing disk {available_unit_number}: datastore_id from config = '{disk_datastore_id}'")
+            LOGGER.info("Processing disk %s: datastore_id from config = '%s'", available_unit_number, disk_datastore_id)
 
             # Check if this disk should use secondary datastore
             if disk_datastore_id == "secondary_datastore_id" and secondary_datastore:
                 disk_datastore = secondary_datastore
                 LOGGER.info(
                     f"Disk {available_unit_number}: Using secondary datastore '{disk_datastore.name}' "
-                    f"(ID: {disk_datastore._moId})"
+                    f"(ID: {disk_datastore._moId})",
```

Ruff (0.14.13)

Logging statement uses f-string

(G004)